text/0000-types-team.md

```diff
@@ -0,0 +1,240 @@
+
+## Relationship to other teams
+
+The types team is conceptually a subteam of both the lang and compiler teams. Since the team repo can't really model that, we'll put it primarily under the lang team.
+
+### Lang team
+
+* Stabilizing new features that intersect the type system will require "types team approval".
+    * Eventually this is expected to require extending the "formality" models to include the feature.
+    * This approval is not meant to be used to make "policy" decisions but to enforce the soundness and implementability of the feature itself.
+* Advising on how to address soundness bugs or other subtle questions that arise.
+* Assisting with the design of language extensions that extends the core Rust type system capabilities
+    * Maintaining the Rust model and extending it to model new proposals that interact with the type checker
+
+### Compiler team
+
+* Assessing and implementing fixes for soundness bugs that have to do with the type system.
+* Maintaining a shared libraries that implements the Rust type/trait system and the borrow checker.
+
+### Initiatives
+
+Initiatives that are extending the semantics of the language, such as the [generic associated types](https://rust-lang.github.io/generic-associated-types-initiative/) initiative, will work closely with this team to integrate those changes into the formality models.
+
+## Examples
+
+Here are some examples to illustrate how the types team will interact.
+
+### Evaluating the implication of equality constraints or negative trait bounds
+
+The original RFC for where clauses included equality constraints like `where T == U` (tracked by [#20041]); similarly, people have regularly considered including the option to have "negated" where clauses like `where T: !Debug`. Both of these features, if led to their full generality, turn out to make the implementation of Rust's trait solver significantly more complicated. Therefore, the types team is within its rights to veto these features or to suggest appropriate modifications to how they work and what they mean. On the other hand, it is up to the lang team to decide the syntax of those where clauses and whether they'd be a useful addition to Rust (presuming they had a semantics the types team was happy with). The types team cannot add a feature to Rust all by itself, but it can either remove one (because it is not feasible to implement) or tweak its formal semantics (to ensure it is sound, feasible etc). In this respect, it is just like the compiler team.
+
+[#20041]: https://github.com/rust-lang/rust/issues/20041
+
+### Bug in the type checker
+
+When a soundness bug is found in Rust's type system, the compiler team can contact the types team to request the bug be triaged and to prepare a fix. The types team owns that code and hence is ultimately responsible for reviewing changes to the relevant code.
```

Comment:
Does the types team make the decision in breaking changes necessary to resolve said soundness bugs?